plugins/gtav/gtav.cpp

```diff
@@ -3,118 +3,213 @@
+#include "ProcessWindows.h"
+#include "mumble_positional_audio_utils.h"
 
-#include <algorithm> // Include algorithm header for the game version detector
+#include <cstring>
 
-static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, float *camera_pos, float *camera_front,
-				 float *camera_top, std::string &context, std::wstring &identity) {
+static std::unique_ptr< ProcessWindows > proc;
+static procptr_t memory_offset = 0;
+static std::string identity;
+
+mumble_error_t mumble_init(uint32_t) {
+	return MUMBLE_STATUS_OK;
+}
+
+void mumble_shutdown() {
+}
+
+MumbleStringWrapper mumble_getName() {
+	static const char name[] = "Grand Theft Auto V";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = name;
+	wrapper.size           = strlen(name);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+mumble_version_t mumble_getAPIVersion() {
+	return MUMBLE_PLUGIN_API_VERSION;
+}
+
+void mumble_registerAPIFunctions(void *) {
+}
+
+void mumble_releaseResource(const void *) {
+}
+
+mumble_version_t mumble_getVersion() {
+	// we reuse the GTA version as plugin version
+	return { 1, 54, 2245 };
+}
+
+MumbleStringWrapper mumble_getAuthor() {
+	static const char author[] = "MumbleDevelopers";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = author;
+	wrapper.size           = strlen(author);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+MumbleStringWrapper mumble_getDescription() {
+	static const char description[] = "Provides positional audio functionality for GTA V. "
+									  "Supports GTA V version 1.54 Build 2245 (Steam) and 1.38 (Retail)."
+									  "Also provides identity based on username.";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = description;
+	wrapper.size           = strlen(description);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+uint32_t mumble_getFeatures() {
+	return MUMBLE_FEATURE_POSITIONAL;
+}
+
+uint8_t mumble_initPositionalData(const char *const *programNames, const uint64_t *programPIDs, size_t programCount) {
+	auto ret                = MUMBLE_PDEC_ERROR_TEMP;
+	constexpr auto exe_name = "GTA5.exe";
+
+	for (size_t i = 0; i < programCount; ++i) {
+		if (strcmp(programNames[i], exe_name) != 0) {
+			continue;
+		}
+
+		proc = std::make_unique< ProcessWindows >(static_cast< procid_t >(programPIDs[i]), programNames[i]);
+		const auto &modules = proc->modules();
+		const auto iter     = modules.find(exe_name);
+		if (iter == modules.cend()) {
+			break;
+		}
+		// find out the offset of the memory region
+		const auto &regions = iter->second.regions();
+		const auto &r       = regions.cbegin();
+		if (r != regions.cend()) {
+			memory_offset = r->address;
+		}
+
+		// Check if we can get meaningful data from it
+		float apos[3], afront[3], atop[3], cpos[3], cfront[3], ctop[3];
+		const char *sidentity = "";
+		const char *scontext  = "";
```

Comment:
```suggestion
		const char *identity = "";
		const char *context  = "";
```

Reply:
Done, note though that `identity` overshadows the global static variable `std::string identity` now, which isn't really pretty.